### PR TITLE
Adding workflow for TPC and ITS time series

### PR DESCRIPTION
**Pull request by matthias-kleiner**

### CODE REVIEW

**Comment by miranov25** [comment]
Hello. 
Is there some progress here? We need this modification for a new tag.

File: DATA/production/configurations/asyncReco/setenv_extra.sh [hunk 1]
```diff
@@ -508,6 +508,12 @@ if [[ $ALIEN_JDL_EXTRACTCURRENTS == 1 ]]; then
   has_detector_reco TPC && add_comma_separated ADD_EXTRA_WORKFLOW "o2-tpc-integrate-cluster-workflow"
 fi
 
+# extra workflows in case we want to process the currents for time series
+if [[ $ALIEN_JDL_EXTRACTTIMESERIES == 1 ]]; then
+  if [[ -z "${WORKFLOW_DETECTORS_RECO+x}" ]] || [[ "0$WORKFLOW_DETECTORS_RECO" == "0ALL" ]]; then export WORKFLOW_DETECTORS_RECO=$WORKFLOW_DETECTORS; fi
+  has_detector_reco TPC && has_detector_reco ITS && has_detector_reco FT0 && add_comma_separated ADD_EXTRA_WORKFLOW "o2-tpc-time-series-workflow"
```
[the remark this answers is not in the document]
Comment (chiarazampolli) [reply DATA/production/configurations/asyncReco/setenv_extra.sh:514]: Hello @matthias-kleiner , 
For the matching, there is the condition `has_detector_matching ITSTPC`. I don't understand the request about `vertex-track-matching-sources`, this is not a workflow, but the track types that one needs for vertexing. Do you mean to set/check `$VERTEXING_SOURCES`?
Did you try the changes in async reco? Before merging, this should be done... 
Cheers,
Chiara

**Comment by chiarazampolli** [comment]
Hello @miranov25 , 
Please, be patient, there are many things ongoing... 
Cheers,
Chiara

File: DATA/production/configurations/asyncReco/setenv_extra.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (matthias-kleiner) [reply DATA/production/configurations/asyncReco/setenv_extra.sh:514]: Thanks, I will try it now locally.
The `vertex-track-matching-sources` was wrong, sorry. I meant the `o2-primary-vertexing-workflow`.

**Comment by matthias-kleiner** [comment]
Technically it was working like it is for me, however when I was adding the check `has_detector_matching ITSTPC` the workflow was not started since the check returned false for me. Not sure if I made something wrong or if I am missing something

**Comment by chiarazampolli** [comment]
Ciao @matthias-kleiner , 

You tried with the async reco script?

We can take a look tomorrow, I won't have many free slots, but we can try to find one.

Chiara

**Comment by shahor02** [comment]
@matthias-kleiner, the defaults for the `WORKFLOW_DETECTORS_...` variables are set in the `$O2DPG_ROOT/DATA/production/setenv.sh`, which is called from the O2 `dpl-workflow.sh`, i.e. *after* the `setenv_extra.sh` is executed.

The only WORKFLOW... settings you have on the  `setenv_extra.sh` level are those which are set by the JDL and parsed by 
the `$O2DPG_ROOT/DATA/production/configurations/asyncReco/async_pass.sh`, which happens to be only `ALIEN_JDL_WORKFLOWDETECTORS` (and if it is not defined, the `WORKFLOW_DETECTORS` is imposed in the `setenv_extra.sh` to have detectors).
Your previous addition of `o2-xxx-integrate-cluster-workflow` is picked up only because you explicitly set `WORKFLOW_DETECTORS_RECO` to `WORKFLOW_DETECTORS`.

For the same reason the `o2-tpc-time-series-workflow` will be picked up in your current version and will be ignored if you require yet undefined ITSTPC matching.

Merging, could you tell which output files should be saved?

**Comment by matthias-kleiner** [comment]
> @matthias-kleiner, the defaults for the `WORKFLOW_DETECTORS_...` variables are set in the `$O2DPG_ROOT/DATA/production/setenv.sh`, which is called from the O2 `dpl-workflow.sh`, i.e. _after_ the `setenv_extra.sh` is executed.
> 
> The only WORKFLOW... settings you have on the `setenv_extra.sh` level are those which are set by the JDL and parsed by the `$O2DPG_ROOT/DATA/production/configurations/asyncReco/async_pass.sh`, which happens to be only `ALIEN_JDL_WORKFLOWDETECTORS` (and if it is not defined, the `WORKFLOW_DETECTORS` is imposed in the `setenv_extra.sh` to have all detectors). Your previous addition of `o2-xxx-integrate-cluster-workflow` is picked up only because you explicitly set `WORKFLOW_DETECTORS_RECO` to `WORKFLOW_DETECTORS`.
> 
> For the same reason the `o2-tpc-time-series-workflow` will be picked up in your current version and will be ignored if you require yet undefined ITSTPC matching.
> 
> Merging, could you tell which output files should be saved?

Thanks for the explanation.
The output file is the `o2_timeseries_tpc.root`.

**Pull request merged**